src/Microsoft.DotNet.Helix/Sdk/HelixTask.cs

```diff
@@ -38,6 +45,10 @@ public sealed override bool Execute()
         {
             try
             {
+                if (IsExternal)
+                {
+                    AnonymousApi = GetHelixApi(requestAnonymous: true);
```

Comment:
I think its easier here to remove the `requestAnonymous` parameter and just do
```suggestion
                    AnonymousApi = ApiFactory.GetAnonymous(BaseUri)
```

Reply:
fair